### PR TITLE
Fix submenu staying open if the mouse leaves the menu

### PR DESCRIPTION
**Pull request by ivanlopezluna**
#### What problem is this solving?
Some cases, when the mouse leaves the parent item, the submenu remains open. This is common if the user moves from the menu to address bar or browser tab, also when it is combined with scrolldown.

#### How to test it?
[Workspace](https://storeframeworkdev--tafmxqa.myvtex.com/)

#### Screenshots or example usage:
**Before**

https://user-images.githubusercontent.com/23274114/187324819-1258d9ad-43fc-4cc8-abbd-d75c5365aec4.mp4

**After**

https://user-images.githubusercontent.com/23274114/187324739-5796c65d-c3d9-41c5-ae14-7ee605c048a7.mp4

#### Related to
PR: https://github.com/vtex-apps/menu/pull/108

### CODE REVIEW

File: react/hooks/useMouseSpeedDebouncer.ts [hunk 1]
```diff
@@ -28,6 +30,8 @@ export const useMouseSpeedDebouncer = (
       const speed = getMouseSpeed()
       if (speed <= maxSpeed) {
         fn(...args)
+      } else if (lastSpeed !== null && lastSpeed >= minUnusualSpeed) {
```
Comment (danzanzini) [inline react/hooks/useMouseSpeedDebouncer.ts:33]: Why not use the maxSpeed also here?
```suggestion
      } else if (lastSpeed !== null && lastSpeed > maxSpeed) {
```

**Review by danzanzini — CHANGES_REQUESTED** [review]
Hey @IvanLopezLuna. Thanks for the contribution. 

Left one comment regarding this new prop. Why not use the maxSpeed instead of a new prop?

File: react/hooks/useMouseSpeedDebouncer.ts (hunk 1 above)
Comment (IvanLopezLuna) [reply react/hooks/useMouseSpeedDebouncer.ts]: Hi @danzanzini! 

You are right, it is a better approach. The change was made, thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+
+### Fixed
+- Fix submenu stays open if mouse leaves menu.
+
```
Comment (filipewl) [inline CHANGELOG.md:11]: Hey @IvanLopezLuna!

Could you please move the added lines under the `## [Unreleased]` line?

**Review by filipewl — CHANGES_REQUESTED** [review]
Looking good @IvanLopezLuna!

I believe @danzanzini's [review](https://github.com/vtex-apps/menu/pull/169#pullrequestreview-1133569142) suggestions have already been addressed. Maybe he just hasn't seen it yet, to remove his "requested changes" block.

I noticed one minor, but important, issue with where the description of the changes was placed in the `CHANGELOG.md` file, so I'm going to leave another change request. 

I'll be watching this PR closely from now on so we can proceed faster!

File: CHANGELOG.md (hunk 2 above)
Comment (IvanLopezLuna) [reply CHANGELOG.md]: Hi @filipewl!

The last requested change it's done.

Thanks for you follow up!

**Review by filipewl — APPROVED** [review]
Thanks for addressing [my review](https://github.com/vtex-apps/menu/pull/169#pullrequestreview-1184413315)! I'm going to approve this now.

However, due to some internal policies related to the Black Friday season, we cannot deploy this right now. Once the permission to deploy is lifted, we'll merge and deploy this. This should happen in the next few days. I'll keep a close eye on this so we can merge as soon as possible. 

Thank you for your patience!

**Review by filipewl — APPROVED** [review]
Deploying this now. Thank you for your patience!

**Pull request merged**